posts/navarroaxel/13.running-graphics-apps-in-docker.md

```diff
@@ -0,0 +1,131 @@
+---
+title: 'Running graphic apps in Docker: AWS WorkSpaces'
+published: true
+description: 'How to run GUI apps using X.Org and Docker in Linux or Windows'
+tags: 'tutorial, docker, aws, security'
+cover_image: ./assets/gui-apps-docker-cover.jpg
+id: 1253471
+---
+
+As a Linux user, I need to connect to an Amazon WorkSpace but the client app can only be run in specific old, but maintained, versions of Ubuntu. I tried with the latest Ubuntu version but it didn't work, so I've created a virtual machine with Ubuntu Focal 20.04 but this setup used 14GB of disk space 😵 only to open a window application. This didn't make any sense. Also this included issues when sharing the clipboard from the host to the VM and then to the virtual desktop hosted in Amazon. 🙄
+Then, I remembered that we can run graphic applications using Docker. Let's see how to connect a container to the [X Window System](https://en.wikipedia.org/wiki/X_Window_System) of the host.
+
+## Creating the Docker image
+
+The following Dockerfile build the image using Ubuntu Focal 20.04, and install the Amazon WorkSpace client:
+
+```Dockerfile
+FROM ubuntu:focal
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt update && \
+  apt install -y libusb-1.0-0 wget xauth && \
+  wget -q -O - https://workspaces-client-linux-public-key.s3-us-west-2.amazonaws.com/ADB332E7.asc | apt-key add - && \
+  echo "deb [arch=amd64] https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu focal main" > /etc/apt/sources.list.d/amazon-workspaces-clients.list && \
+  apt remove -y wget && \
+  apt update && \
+  apt install -y workspacesclient && \
+  touch /root/.Xauthority && \
+  rm -rf /var/lib/apt/lists/*
+```
+
+💡 `libusb-1.0-0` is a dependency of Amazon WorkSpaces.
```

Comment:
Why not "💡 `libusb-1.0-0` is an Amazon WorkSpaces' dependency"? 🤔

Reply:
It installed by default in the Linux Desktop Environment (GNOME, KDE Plasma, etc.). So, the bug is in the WorkSpaces `.deb` file. They should mark this lib as a dependency, but this bug is not present in any graphic Ubuntu installation.